apps/ui/middleware.ts

```diff
@@ -29,6 +31,40 @@ function redirectProtectedRoute(req: NextRequest) {
   return undefined;
 }
 
+/// are we missing a csrf token in cookies and should redirect to get one
+function redirectCsrfToken(req: NextRequest): URL | undefined {
+  // need to ignore all assets outside of html requests (which don't have an extension)
+  // so could we just check any request that doesn't have an extension?
+  if (
```

Comment:
Would it work to limit redirects to paths matching apiBaseUrl + /graphql/ and /v1/ for REST?